qemu/qmp_test.go

```diff
@@ -1504,3 +1519,47 @@ func TestExecuteNVDIMMDeviceAdd(t *testing.T) {
 	q.Shutdown()
 	<-disconnectedCh
 }
+
+func TestMainLoopEventBeforeGreeting(t *testing.T) {
+	connectedCh := make(chan *QMPVersion)
+	disconnectedCh := make(chan struct{})
+	buf := newQMPTestCommandBufferNoGreeting(t)
+
+	// register a channel to receive events
+	eventCh := make(chan QMPEvent)
+	cfg := QMPConfig{EventCh: eventCh, Logger: qmpTestLogger{}}
+	q := startQMPLoop(buf, cfg, connectedCh, disconnectedCh)
```

Comment:
Could we modify the test case to wait for the version string before starting the event loop.  If I'm correct this will deadlock with the current code.  Moving the break out of the if statement should fix the issue.  BTW there is some utility test code for in this file for testing events.  Please see TestDelEvent.

Reply:
ok, got it